src/TeamCloud.Orchestrator/Orchestrations/Azure/AzureResourceGroupDeleteActivity.cs

```diff
@@ -0,0 +1,62 @@
+﻿/**
+ *  Copyright (c) Microsoft Corporation.
+ *  Licensed under the MIT License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Rest.Azure;
+using TeamCloud.Azure;
+using TeamCloud.Model.Data;
+
+namespace TeamCloud.Orchestrator.Orchestrations.Azure
+{
+    public class AzureResourceGroupDeleteActivity
+    {
+        private readonly IAzureSessionService azureSessionService;
+
+        public AzureResourceGroupDeleteActivity(IAzureSessionService azure)
+        {
+            this.azureSessionService = azure ?? throw new ArgumentNullException(nameof(azure));
+        }
+
+        [FunctionName(nameof(AzureResourceGroupDeleteActivity))]
+        public async Task RunActivity(
+            [ActivityTrigger] AzureResourceGroup azureResourceGroup)
+        {
+            if (azureResourceGroup == null)
+                throw new ArgumentNullException(nameof(azureResourceGroup));
+
+            try
+            {
+                var azure = azureSessionService.CreateSession(azureResourceGroup.SubscriptionId);
+                var locks = await azure.ManagementLocks.ListByResourceGroupAsync(azureResourceGroup.ResourceGroupName, true).ConfigureAwait(false);
+                var ids = locks.Select(s => s.Id).ToArray();
+                if (ids.Length > 0)
+                {
+                    await azure.ManagementLocks.DeleteByIdsAsync(ids).ConfigureAwait(false);
+                    // Wait one minute after deleting locks to ensure that Azure has updated itself before trying to delete the RG
+                    await Task.Delay(60 * 1000).ConfigureAwait(false);
```

Comment:
no way - we can't rely on a delay here. what if we have to clean up 100 locks. does this take longer?

Reply:
This was replaced with a loop.  Loop will abort after 10 tries to make sure it's not ever permanently stuck in a loop.